Library/Homebrew/os/mac/sdk.rb

```diff
@@ -91,6 +91,12 @@ def sdk_paths
               paths[OS::Mac::Version.new(version)] = sdk_path if version.present?
             end
 
+            if OS::Mac.version >= :big_sur
```

Comment:
Should/could there be any logic put or changed in `sdk_if_applicable` too? I just note that that already references `:big_sur`. Additionally, a comment would be good here to note why this is being done.

Reply:
>Should/could there be any logic put or changed in `sdk_if_applicable` too

I'll have a think about it. What did you have in mind?

>a comment would be good here to note why this is being done

I've been thinking exactly that too. I'll put one in.

Reply:
> I'll have a think about it. Did you have anything in particular in mind?

Nothing specifically! Just wondering if the logic should/could be combined. If you or others think "nope!": fine with me.

Reply:
Hmmm, no, I think it's fine to keep it. If I were to make a change to it, I would actually just remove the line involving the `:big_sur` check

https://github.com/Homebrew/brew/blob/993bf95877143afe1cede022083499e0b4fe336c/Library/Homebrew/os/mac/sdk.rb#L57-L64

as I _think_ it's essentially a no-op after this PR. 

However, this is under the assumption that `sdk_prefix` always contains a `MacOSX.sdk` that points to the right version. We'd need to put it back if this stops being true, or remove/modify the change in this PR (and possibly elsewhere) if `MacOSX.sdk` suddenly starts pointing to an SDK that we _don't_ want.

There are probably other ways to combine the logic, but I don't see a clean way of doing it right now. I'd need to invest more time in understanding the code to figure it out.

Given the uncertainty regarding the contents of `sdk_prefix`, I think it's best to keep the rest of the code as-is for now. We can revisit it when we're a few minor upgrades into Big Sur and have a better understanding of how Apple wants to do things.

Reply:
> sdk_prefix always contains a MacOSX.sdk that points to the right version

I was wondering about that. I don't know if there are versions of Xcode/CLT for which it could be untrue.

Reply:
> However, this is under the assumption that `sdk_prefix` always contains a `MacOSX.sdk` that points to the right version. We'd need to put it back if this stops being true, or remove/modify the change in this PR (and possibly elsewhere) if `MacOSX.sdk` suddenly starts pointing to an SDK that we _don't_ want.

Could this be a fallback so if it isn't set at all we set it to this logic here?

I'd be tempted to remove this logic if it becomes a no-op after this but hopefully @mistydemeo will be able to chime in with thoughts.

I'll merge this as-is or in some form tomorrow when other maintainers have (hopefully) had the chance to take a look. 

Thanks @carlocab!

Reply:
>it becomes a no-op

To be clear, I think it (the pre-existing `:big_sur` check) will do two things when this PR is merged:
1. it skips the version check in the succeeding line; and,
2.  handles the case where `MacOSX.sdk` disappears for some reason _and_ there is a minor-version mismatch in Big Sur's SDK.

But, yes, I would also be interested in other opinions on this to be sure.

Reply:
> 2\. handles the case where `MacOSX.sdk` disappears for some reason

Unless we've seen cases where it isn't present: I think it's worth the assumption that this won't happen on Big Sur and above. Thoughts?

Reply:
I am inclined to agree, but I'll admit I haven't exactly been paying attention to the contents of `sdk_prefix`. I also avoid installing Xcode like the plague, so I know even less about `sdk_prefix` for Xcode.

Reply:
@Bo98 any thoughts on any of this?